mirgecom/__init__.py

```diff
@@ -25,26 +25,23 @@
 import sys
 from warnings import warn
 
-try:
-    from mpi4py import MPI
+from mpi4py import MPI
```

Comment:
We shouldn't unconditionally import `mpi4py`. Running mirgecom sequentially without MPI is entirely legitimate IMO (but let's chat about that in the dev meeting tomorrow, added to agenda).

I get that that makes parallel runs tougher to detect, but, as a fallback not requiring MPI, we could also fish around for some env vars set by a queueing system for evidence of large runs.

Reply:
I think the consensus in the meeting was to always require mpi4py?

Reply:
Yes, agreed.